src/components/Calculator.js

```diff
@@ -1,125 +1,151 @@
 import React from "react";
 import "./Layout.css"
+import calculate from '../logic/calculate';
 class Structure extends React.Component {
   constructor(props){
     super(props)
+    this.state = {
+      total: 0,
+      next: '',
+      operation: '',
+    }
+    this.handleClick = this.handleClick.bind(this)
   }
+
+  handleClick = (event) => {
+    const buttonName = event.target.value;
+    this.setState((prevState) => calculate(prevState, buttonName));
+  }
+
   render(){
   return (
     <div className="frame">
       <div className="calculator">
         <br />
-        <div>
-          <input
-            type="text"
-            readOnly
-            className="screen"
-            style={{ textSize: "40px" }}
-            value={0}
-          />
+        <div> 
+        <button className="screen">{this.state.total}{this.state.operation}{this.state.next}</button>      
         </div>
         <div className="keys">
           <input
             type="button"
             value="AC"
             className="button-clear"
+            onClick={this.handleClick}
           />
           <input
             type="button"
             value="+/-"
             className="button-operator"
+            onClick={this.handleClick}
           />
           <input
             type="button"
             value="%"
             className="button-operator"
+            onClick={this.handleClick}
           />
           <input
             type="button"
             value="&#247;"
             className="button-operator"
             id="red"
+            onClick={this.handleClick}
           />
           <input
             type="button"
             value="7"
             className="button-operator"
+            onClick={this.handleClick}
           />
           <input
             type="button"
             value="8"
             className="button-operator"
+            onClick={this.handleClick}
           />
           <input
             type="button"
             value="9"
             className="button-operator"
+            onClick={this.handleClick}
           />
           <input
             type="button"
             value="*"
             className="button-operator"
             id="red"
+            onClick={this.handleClick}
           />
```

Comment:
- [ ] I'm not able to perform multiplication operations, When I do the applications crash. Please fix this issue and make sure the users are able to perform all maths operations.
  - This is happening because in `oprate.js` file line 12 it's `x` for multiplication and here on this input you are using `*`.
  - To fix this, On this button replace the value `*` with `x`.

Reply:
Thank you so much @thecodechaser .